sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ItemsPartitionReader.scala

```diff
@@ -138,6 +142,15 @@ private case class ItemsPartitionReader
         ).toInt
       )
 
+      readConfig.maxIntegratedCacheStaleness match {
```

Comment:
this code is duplicated in several places - please add a help method that will convert the maxINtegratedCacheStaleness into new DedicatedGatewayRequestOptions in the readConfig itself and reuse it wherever this conversion is happening